ingestion/src/metadata/ingestion/source/database/iceberg/models.py

```diff
@@ -51,24 +51,29 @@ def from_pyiceberg(
         """Responsible for parsing the needed information from a PyIceberg Table."""
         iceberg_columns = table.schema().fields
 
+        partition_columns = []
+        for partition in table.spec().fields:
+            parition_column_name = get_column_from_partition(iceberg_columns, partition)
+            column_parition_type = get_column_partition_type(iceberg_columns, partition)
+
+            if not (parition_column_name and column_parition_type):
+                continue
+
+            partition_columns.append(
+                PartitionColumnDetails(
+                    columnName=parition_column_name,
+                    intervalType=column_parition_type,
+                    interval=None,
+                )
+            )
+
         return IcebergTable(
             name=name,
             tableType=table_type,
             description=table.properties.get("comment"),
             owners=owners,
             columns=[IcebergColumnParser.parse(column) for column in iceberg_columns],
-            tablePartition=TablePartition(
-                columns=[
-                    PartitionColumnDetails(
-                        columnName=get_column_from_partition(
-                            iceberg_columns, partition
-                        ),
-                        intervalType=get_column_partition_type(
-                            iceberg_columns, partition
-                        ),
-                        interval=None,
-                    )
-                    for partition in table.spec().fields
-                ]
-            ),
+            tablePartition=partition_columns
+            and TablePartition(columns=partition_columns)
+            or None,
```

Comment:
let's keep if-else condition here for better readability.
```suggestion
            tablePartition = TablePartition(columns=partition_columns) if partition_columns else None
```

Reply:
done